File: coderbyte/a_easy/24_swap_case/swap_case_custom.java
class Main {

  /**
  * This is a reimplemented SwapCase method in Java.
  * Compared to regular SwapCase method, it takes a len parameter.
  *
  * @author togiberlin
  */

  public static void main(String[] args) {
    System.out.println(swapCase("Hello World!", 6));
  }

  public static String swapCase(String str, int len) {
    char[] charArray = str.toCharArray();

    for (int i = 0; i < len; i++) {
      char character = charArray[i];

      if (Character.isUpperCase(character)) {
        charArray[i] = Character.toLowerCase(character);
      } else if (Character.isLowerCase(character)) {
        charArray[i] = Character.toUpperCase(character);
      }
    }

    return new String(charArray);
  }
}

File: coderbyte/a_easy/javascript/24_swap_case/swap_case_custom.js
const swapCase = (string, offset) => {
  const array = string.split('');

  for (let i = 0; i < offset; i++) {
    if (array[i].match(/[a-z]/)) {
      array[i] = array[i].toUpperCase();
    } else if (array[i].match(/[A-Z]/)) {
      array[i] = array[i].toLowerCase();
    }
  }

  return array.join('');
};

module.exports = swapCase;
